src/proxy_cli.rs

```diff
@@ -25,22 +28,61 @@ pub fn proxy_run(arg0: &str) -> Result<ExitCode> {
 }
 
 fn direct_proxy(proc_name: &str, args: &[OsString], toolchain: &Toolchain) -> io::Result<ExitCode> {
-    let bin_path = toolchain.bin_path.join(proc_name);
+    let toolchain_override: Option<ToolchainOverride> = ToolchainOverride::from_project_root();
+
+    let (bin_path, toolchain_name) = match toolchain_override {
+        Some(to) => {
+            // unwrap() is safe here since we try DistToolchainDescription::from_str()
+            // when deserializing from the toml.
+            let description =
+                DistToolchainDescription::from_str(&to.cfg.toolchain.channel).unwrap();
+
+            // Since we have a valid
+            let toolchain = Toolchain::from_path(&description.to_string()).unwrap_or_else(|_| {
+                panic!("Failed to create toolchain '{}' from path", &description)
```

Comment:
Note: I'm using `panic!` here and below for 2 reasons:

1) this is the point of entry into fuelup-managed commands like `forc`, so we shouldn't continue with execution if we end up with bad state here, and
2) for this particular fn, creating toolchain from path that we've already done some validation above shouldn't fail.

Reply:
> this is the point of entry into fuelup-managed commands like forc, so we shouldn't continue with execution if we end up with bad state here

This doesn't imply we should `panic!`, but rather that we should make sure when this error is raised that we actually exist the program (and don't try to continue).

> for this particular fn, creating toolchain from path that we've already done some validation above shouldn't fail.

This on the other hand is a good reasons to `panic!`, i.e. if we're checking something that we *know* has already been checked.

That said, it looks like this `Toolchain` constructor does `I/O` (reads from a file), so it doesn't really matter if we've already checked it before - it's always possible that since the last time we checked it, it's been deleted or corrupted by some other process or anything really, so technically we should probably handle this.

This is not a huge deal for fuelup, but just wanted to clarify good practise around producing errors vs `panic!`ing.

Reply:
Actually I think there's a simpler solution - there wasn't any reason to return a `Result<Toolchain>` anyway since there wasn't anything upstream that returns a `Result`. Will create another PR to update the fn signature to return `Self` instead of `Result<Self>` (so we can keep things clean here)

I updated the `direct_proxy` fn signature to return `anyhow::Error` over `io::Error` (downstream `exec()` fn still returns io::Result), this probably makes things easier.

But definitely good points regarding `panic!` vs errors, will keep in mind :+1: